pandas/tests/series/indexing/test_indexing.py

```diff
@@ -523,7 +523,22 @@ def test_setitem_with_tz_dst():
     tm.assert_series_equal(s, exp)
 
 
-def test_categorial_assigning_ops():
+def test_setitem_new_key_tz():
```

Comment:
there is a file for timezones and timeseries tests under indexing

prefer one of those

Reply:
sure.  for next time, when does a test go in tests.indexing vs tests.series.indexing vs tests.frame.test_indexing?

Reply:
both tests.series.indexing.test_datetime and tests.indexing.test_datetime are about objects with DatetimeIndex, not datetime values

Reply:
well test_indexing is really big, just like to split out when dealing with specific types (may not be perfect but that's the idea); this should clearly be there.

Reply:
i think figuring this out merits a dedicated pass(es) similar to tests.arithmetic and tests.reductions

Reply:
maybe, but better to put this where it should go now

Reply:
this file is only 842 lines long and contains a bunch of tests for setitem.  I still don't see what the preferred alternative is

Reply:
see my comment below, I am sure we test this for .loc, so this belongs with it.

Reply:
OK.  Do you have a suggestion as to where that might be?  I'm not eager to manually sort through 50k+ tests.  (and there aren't _any_ `foo.loc` setitem calls in either tests.series.test_timezones or tests.series.test_timeseries)

Reply:
``test_loc_coerceion`` in pandas/tests/indexing/test_loc.py

Reply:
everything in that file is testing loc.  you want to mix non-loc tests in with that?

Reply:
moved